ir/visitor.cpp

```diff
@@ -416,3 +417,49 @@ ControlFlowVisitor &ControlFlowVisitor::flow_clone() {
     assert(rv->check_clone(this));
     return *rv;
 }
+
+#define DEFINE_APPLY_FUNCTIONS(CLASS, TEMPLATE, TT)                                             \
+    TEMPLATE bool IR::CLASS TT::apply_visitor_preorder(Modifier &v)                      \
```

Comment:
They're actually not identical, alas, because one macro includes the `inline` keyword and the other doesn't. I can set up something to eliminate the duplication, though.